api/api/accounts/views.py

```diff
@@ -1,38 +1,36 @@
 from django.contrib.auth import get_user_model
-from rest_framework import status
-from rest_framework.generics import CreateAPIView, RetrieveUpdateAPIView
+from rest_framework import generics
 from rest_framework.permissions import IsAdminUser
-from rest_framework.response import Response
-from rest_framework.views import APIView
 
 from api.roles.models import Roles
-from api.roles.serializers import RolesSerializer
+from api.roles.serializers import RoleSerializer
 
 from .permissions import IsOwnerOrAdmin
 from .serializers import UserSerializer
 
 User = get_user_model()
 
 
-class CreateUserView(CreateAPIView):
+class CreateUserView(generics.CreateAPIView):
     """ View to create users """
     permission_classes = (IsAdminUser,)
     queryset = User.objects.all()
     serializer_class = UserSerializer
 
 
-class RetrieveUpdateUserView(RetrieveUpdateAPIView):
+class RetrieveUpdateUserView(generics.RetrieveUpdateAPIView):
     """ View to retrieve and update user """
     permission_classes = (IsOwnerOrAdmin,)
     queryset = User.objects.all()
     serializer_class = UserSerializer
 
 
-class ListUserRoles(APIView):
-    def get(self, request, pk):
-        queryset = User.objects.get(pk=pk).roles.filter(
+class ListUserRoles(generics.ListAPIView):
+    serializer_class = RoleSerializer
+    lookup_url_kwarg = "pk"
+
+    def get_queryset(self):
+        pk = self.kwargs.get(self.lookup_url_kwarg)
+        user_roles = User.objects.get(pk=pk).roles.filter(
```

Comment:
What happens when the user doesn't exist?

Reply:
We are using
` 'DEFAULT_PERMISSION_CLASSES': (
        'rest_framework.permissions.IsAuthenticated',
    )`
   
So user will be a valid user